subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestFramework.java

```diff
@@ -0,0 +1,88 @@
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.testing.junitplatform;
+
+import org.gradle.api.Action;
+import org.gradle.api.internal.tasks.testing.TestClassProcessor;
+import org.gradle.api.internal.tasks.testing.TestFramework;
+import org.gradle.api.internal.tasks.testing.WorkerTestClassProcessorFactory;
+import org.gradle.api.internal.tasks.testing.detection.ClassFileExtractionManager;
+import org.gradle.api.internal.tasks.testing.detection.TestFrameworkDetector;
+import org.gradle.api.internal.tasks.testing.filter.DefaultTestFilter;
+import org.gradle.api.tasks.testing.Test;
+import org.gradle.api.tasks.testing.junitplatform.JUnitPlatformOptions;
+import org.gradle.internal.UncheckedException;
+import org.gradle.internal.actor.ActorFactory;
+import org.gradle.internal.id.IdGenerator;
+import org.gradle.internal.service.ServiceRegistry;
+import org.gradle.internal.time.Clock;
+import org.gradle.process.internal.worker.WorkerProcessBuilder;
+
+import java.io.Serializable;
+import java.lang.reflect.Constructor;
+
+public class JUnitPlatformTestFramework implements TestFramework {
+    private JUnitPlatformOptions options;
+    private final JUnitPlatformDetector detector;
+    private final DefaultTestFilter filter;
+
+    public JUnitPlatformTestFramework(Test testTask, DefaultTestFilter filter) {
+        this.filter = filter;
+        options = new JUnitPlatformOptions();
+        detector = new JUnitPlatformDetector(new ClassFileExtractionManager(testTask.getTemporaryDirFactory()));
+    }
+
+    @Override
+    public WorkerTestClassProcessorFactory getProcessorFactory() {
+        return new JUnitPlatformTestClassProcessorFactory();
+    }
+
+    @Override
+    public Action<WorkerProcessBuilder> getWorkerConfigurationAction() {
+        return new Action<WorkerProcessBuilder>() {
+            public void execute(WorkerProcessBuilder workerProcessBuilder) {
+                workerProcessBuilder.sharedPackages("org.junit.platform");
+            }
+        };
+    }
+
+    @Override
+    public JUnitPlatformOptions getOptions() {
+        return options;
+    }
+
+    @Override
+    public TestFrameworkDetector getDetector() {
+        return detector;
+    }
+
+    public static class JUnitPlatformTestClassProcessorFactory implements WorkerTestClassProcessorFactory, Serializable {
+        @Override
+        public TestClassProcessor create(ServiceRegistry serviceRegistry) {
+            try {
+                IdGenerator idGenerator = serviceRegistry.get(IdGenerator.class);
+                Clock clock = serviceRegistry.get(Clock.class);
+                ActorFactory actorFactory = serviceRegistry.get(ActorFactory.class);
+                Class clazz = getClass().getClassLoader().loadClass("org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor");
+                Constructor constructor = clazz.getConstructor(IdGenerator.class, Clock.class, ActorFactory.class);
```

Comment:
:x: The order of constructor parameters is incorrect. It should be `ActorFactory` and then `Clock`. If I run the current code with a Gradle distribution on a real project I get the following exception:

```
org.gradle.internal.UncheckedException: java.lang.NoSuchMethodException: org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.<init>(org.gradle.internal.id.IdGenerator, org.gradle.internal.time.Clock, org.gradle.internal.actor.ActorFactory)
        at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:63)
        at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:40)
        at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestFramework$JUnitPlatformTestClassProcessorFactory.create(JUnitPlatformTestFramework.java:84)
        at org.gradle.api.internal.tasks.testing.worker.TestWorker.startReceivingTests(TestWorker.java:83)
        at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:66)
        at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:45)
        at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:83)
        at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:35)
        at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:119)
        at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:64)
        at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:62)
        at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:67)
Caused by: java.lang.NoSuchMethodException: org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.<init>(org.gradle.internal.id.IdGenerator, org.gradle.internal.time.Clock, org.gradle.internal.actor.ActorFactory)
        at java.base/java.lang.Class.getConstructor0(Class.java:3302)
        at java.base/java.lang.Class.getConstructor(Class.java:2110)
        at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestFramework$JUnitPlatformTestClassProcessorFactory.create(JUnitPlatformTestFramework.java:81)
        ... 9 more
```

Given that we do not receive the exception in your tests in Gradle core makes me believe that your Spock extension doesn't work properly.

Reply:
Sorry... I make all integration tests pass then slightly changed the parameter order. Anyway, now I pushed it to CI.

Reply:
If I don't define `org.apiguardian:apiguardian-api:1.0.0` in my build script then I get messages like:

```
warning: unknown enum constant Status.STABLE
  reason: class file for org.apiguardian.api.API$Status not found
```

Seems like the dependency is not picked up properly.

Reply:
We should also add samples to the user guide that show the usage from the perspective of an actual build script. Those samples also need to be tested.